pkg/plugin/v1/api_flags.go

```diff
@@ -0,0 +1,96 @@
+// Copyright 2020 The Operator-SDK Authors
```

Comment:
IMO, we need to extract the apiflags (api_flags.go):

Reply:
The only shared flags would be `group`, `version`, `kind`, and `crd-version`. If/when we want to split the implementations out of the SDK, it would require that we put the code for these 4 flags into a separate shared repo (and make them exported) so that both ansible and helm operator repos could import and use it.

IMO, there's not a whole lot of value in sharing these four flags and dealing with all of the dependency/versioning issues that would come along with it.

Reply:
Hi @joelanford, 

I did not write properly. the reason for that is for both plugin's command api and create api. 
We cannot call the create api validation in the init because the operation to create an API has diff validations. For example, when we create the api we should check that the config exists and the plugin is helm when we are calling the init not.

Reply:
> We cannot call the create api validation in the init because the operation to create an API has diff validations. For example, when we create the api we should check that the config exists and the plugin is helm when we are calling the init not.

Per a previous comment, I don't think this is correct. The Kubebuilder CLI will do this type of validation for us (and in fact _HAS_ to do it to know which plugin to load during `create api`).

Therefore I think the `create api` validations _WE_ need to do are the same in both `init` and `create api`, and at that point, I would again probably say that the use of this extra `APIFlags` struct is unnecessary and overly complex.